### ISSUE TITLE
Modify ResisdualsPlot histogram to optionally use a PDF

### ISSUE THREAD

**Issue by bbengfort** [DistrictDataLabs/yellowbrick#523]
Because the training data set is usually much larger than the test data set, adding an option to normalize the histogram by using a PDF instead of pure frequency may make comparing the train and test set residuals distribution easier. 

Right now the histogram is drawn on the plot if the argument `hist=True`; we can change this argument to accept a string or a boolean, e.g. `hist=True` or `hist='density'` draws the PDF, `hist='frequency'` draws the normal histogram, and `hist=None` or `hist=False` does not draw the histogram at all. Alternatively, we can simply add another boolean argument, `density=True` to the visualizer if that is more understandable. 

The histogram is drawn on [line 508](https://github.com/DistrictDataLabs/yellowbrick/blob/develop/yellowbrick/regressor/residuals.py#L507), to use the PDF, pass `density=True` to the `hist()` function.
